File: hugin/example/02-plot3d/plot3d.py
import numpy as np
import matplotlib.pyplot as plt
from mpl_toolkits.mplot3d import Axes3D

def create_helix_data():
    # Create 100 points from 0 to 4π
    t = np.linspace(0, 4 * np.pi, 100)
    # x = cos(t), y = sin(t), z = t / (4π)
    x = np.cos(t)
    y = np.sin(t)
    z = t / (4 * np.pi)  # Linear z scaled from 0 to 1
    return x, y, z

def main():
    # Generate data
    x, y, z = create_helix_data()
    print("x:", x)
    print("\n")
    print("y:", y)
    print("\n")
    print("z:", z)
    print("\n")
    
    # Create a figure and 3D axes
    fig = plt.figure(figsize=(8, 6))
    ax = fig.add_subplot(111, projection='3d')
    
    # Plot the 3D helix
    ax.plot(x, y, z, color='b', linewidth=1.0)
    
    # Set labels and title
    ax.set_title("3D Helix")
    ax.set_xlabel("x")
    ax.set_ylabel("y")
    ax.set_zlabel("z")
    
    # Display the plot
    plt.show()

if __name__ == "__main__":
    main()

File: hugin/example/02-plot3d/plot3d.ml
open Nx

let create_helix_data () =
  let t = linspace float32 0. (4. *. Float.pi) 100 in
  let x = cos t in
  let y = sin t in
  let z = map_item (fun t -> t /. (4. *. Float.pi)) t in
  (x, y, z)

let () =
  let x, y, z = create_helix_data () in
  let fig =
    Hugin.plot3d ~title:"3D Helix" ~xlabel:"x" ~ylabel:"y" ~zlabel:"z" x y z
  in
  Hugin.show fig
